cypress/e2e/tests/pages/user_menu/preferences.spec.ts

```diff
@@ -0,0 +1,368 @@
+    };
+
+    PagePo.goTo(endpoint);
+    for (const [key, value] of Object.entries(languages)) {
+      prefPage.dropdownMenu().open(0);
+      prefPage.listBox().isOpened();
+      prefPage.listBox().getListBoxItems().should('have.length', 2);
+      prefPage.listBox().set(key);
+      prefPage.dropdownMenu().checkOptionSelected(0, key);
+      prefPage.listBox().isClosed();
+      prefPage.checkLangDomElement(value);
+      cy.reload();
+      prefPage.dropdownMenu().checkOptionSelected(0, key);
+      prefPage.checkLangDomElement(value);
+    }
+  });
+
+  it('Can select a theme', () => {
+    /*
+    Select theme and verify that its highlighted
+    Verify selection and mode ('Light' or 'Dark') is preserved after logout/login
+    */
+    const themeOptions = {
+      Light: 'theme-light',
+      Dark:  'theme-dark',
+      Auto:  ''
+    };
+    // Set theme for 'Auto' mode based on time of day
+    const hour = new Date().getHours();
+
+    for (const key in themeOptions) {
+      (key === 'Auto' && hour < 7 || hour >= 18) ? themeOptions['Auto'] = 'theme-dark' : themeOptions['Auto'] = 'theme-light';
+    }
+
+    PagePo.goTo(endpoint);
+    for (const [key, value] of Object.entries(themeOptions)) {
+      PagePo.goTo(endpoint);
+      prefPage.button(key).set();
+      prefPage.button(key).isSelected();
+      prefPage.checkThemeDomElement(value);
+      userMenu.open();
+      userMenu.checkOpen();
+      userMenu.clickMenuLink('Log Out');
+      cy.login();
+      PagePo.goTo(endpoint);
+      prefPage.button(key).isSelected();
+      prefPage.checkThemeDomElement(value);
+    }
+  });
+
+  it('Can select login landing page', () => {
+    /*
+    Select each radio button and verify its highlighted
+    Verify user is landing on correct page after login
+    Verify selection is preserved after logout/login
+    */
+    const landingPageOptions: {[key: number]: string} = {
+      0: '/home',
+      1: 'c/_/manager/provisioning.cattle.io.cluster',
+      // 2: '/explore' // TODO this option only works when there is an existing cluster (not for standard user)
+    };
+
+    for (const [key, value] of Object.entries(landingPageOptions)) {
+      PagePo.goTo(endpoint);
+      prefPage.radioButton().set(key);
+      prefPage.radioButton().isChecked(key);
+
+      // if key is 1, navigate to cluster manager page and then do validations, else just do validations
+      if (key == 1) {
+        cy.intercept('GET', '/v3/clusters').as('clusters');
+        PagePo.goTo('c/_/manager/provisioning.cattle.io.cluster');
+        cy.wait('@clusters').its('response.statusCode').should('eq', 200);
+      }
+
+      userMenu.open();
+      userMenu.checkOpen();
+      userMenu.clickMenuLink('Log Out');
+      cy.login();
+      cy.visit(Cypress.config().baseUrl);
+      cy.url().should('include', value);
+      PagePo.goTo(endpoint);
+      prefPage.radioButton().isChecked(key);
+    }
+  });
+
+  it('Can select date format', () => {
```

Comment:
this is just really validating the LabelSelect. in places where we can't easily check screen content we should correct http request is sent (v1/userpreferences/user-<id> PUT payload contains the correct `date-format` value and has an ok responce)

Reply:
updated here: https://github.com/rancher/dashboard/pull/8705/commits/c977db38f17847cb67be5cf27d47e413678f3f00